Sources/MapboxNavigation/SpriteRepository.swift

```diff
@@ -4,16 +4,13 @@ import MapboxCoreNavigation
 import MapboxDirections
 
 class SpriteRepository {
-    // Caching the Sprite and single shield icon images.
-    let spriteCache = ImageCache()
-    // Caching the single legacy shield icon images.
-    let legacyCache = ImageCache()
-    // Caching the metadata info for Sprite.
-    let infoCache =  SpriteInfoCache()
     let baseURL: URL = URL(string: "https://api.mapbox.com/styles/v1")!
     var styleURI: StyleURI = .navigationDay
-    fileprivate(set) var imageDownloader: ReentrantImageDownloader = ImageDownloader()
+    fileprivate(set) var requestDownloader: ReentrantRequestDownloader = RequestDownloader()
+    private let headers: [String: String] = ["Accept": "image/*;q=0.8"]
     
+    let requestCache: BimodalURLCache = URLDataCache()
```

Comment:
Use the requestCache to store downloaded Sprite image, Sprite Info and single legacy shield. The derivedCache is used to store the generated shield, exit views and the single sprite shield icon.